components/datetime/src/skeleton.rs

```diff
@@ -91,6 +91,20 @@ impl<'de> de::Visitor<'de> for DeserializeSkeletonBincode {
     {
         let mut items: SmallVec<[fields::Field; 5]> = SmallVec::new();
         while let Some(item) = seq.next_element()? {
+            if let Some(prev_item) = items.last() {
+                if prev_item > &item {
+                    return Err(de::Error::invalid_value(
+                        de::Unexpected::Other(&format!("field item out of order: {:?}", item)),
+                        &"ordered field symbols representing a skeleton",
+                    ));
+                }
+                if prev_item == &item {
+                    return Err(de::Error::invalid_value(
+                        de::Unexpected::Other(&format!("duplicate field: {:?}", item)),
+                        &"ordered field symbols representing a skeleton",
+                    ));
+                }
```

Comment:
Nitpick (optional): combine error handling via `prev_item >= &item`.  Less code to read and compile for an error that shouldn't happen in normal situations.

Reply:
Since everything is green, I may fast follow with this one.